File: dart/problem029.dart
import 'dart:collection';
import 'dart:math';

import 'euler_solution.dart';

main(List<String> args) {
  EulerSolution.execute(new Problem029());
}

class Problem029 extends EulerSolution {
  @override
  int run() {
    var terms = new HashSet<int>();

    for (var a = 2; a <= 100; a++) {
      for (var b = 2; b <= 100; b++) {
        int big = pow(a, b);
        terms.add(big);
      }
    }

    return terms.length;
  }
}

File: ruby/problem029.rb
require './euler_solution.rb'
require 'set'

class Problem029
  include EulerSolution

  def run
    terms = Set.new
    (2..100).each do |a|
      (2..100).each do |b|
        terms << a ** b
      end
    end
    terms.size
  end
end

execute(Problem029.new)
